bms.h

```diff
@@ -0,0 +1,58 @@
+#ifndef __BMS_H_
+#define __BMS_H_
+
+#include <vector>
+#include <string>
+
+class Battery{
+  public:
+    const float TEMPERATURE_LOWER_BOUNDARY = 0;
+    const float TEMPERATURE_UPPER_BOUNDARY = 45;
+    const float SOC_LOWER_BOUNDARY = 20;
+    const float SOC_UPPER_BOUNDARY = 80;
+    const float CHARGE_RATE_LOWER_BOUNDARY = 0.0;
+    const float CHARGE_RATE_UPPER_BOUNDARY = 0.8;
+
+    typedef enum BMS_PARAMETER
+    {
+        BATTERY_TEMPERATURE = 0,
+        BATTERY_SOC,
+        BATTERY_CHARGE_RATE
+    };
+
+    typedef enum BREACH_TYPE
+    {
+       LOW_BREACH = 0,
+       HIGH_BREACH,
+       OUT_OF_RANGE
+    };
+
+    typedef enum CRITICALITY{
+      WARNING,
+      ERROR
+    };
+
+    typedef enum LANGUAGE{
+      ENGLISH = 0,
+      GERMAN,
+    };
+
+    const int logLanguage = ENGLISH; /*for language change change enum*/
+    //const int logLanguage = GERMAN;
+
+    std::vector<std::vector<std::string>> batteryParameter = { { "Temperature", "State of charge", "Charge Rate" }, {"Temperatur", "Ladezustand", "Ladestrom"} };
+    std::vector<std::vector<std::string>> batteryBreach = {{ "Low", "High", "Out Of Range" } ,{"niedrig", "hoch", "Au�er Reichweite"}};
+    std::vector<std::vector<std::string>> batteryBreachCriticality = {{ "Warning", "Error" },{"Warnung","Error"}};
+};
+
+class BatteryManagementSystem : public Battery{

```

Comment:
There was a method BatteryIsOk() which is a single API that provides the status which is a function of all parameters.
By not having such method, the user is expected to know each parameter and verify each parameter separately. Please see if not having that method is convenient or not.